semhash/datamodels.py

```diff
@@ -102,6 +103,30 @@ def rethreshold(self, threshold: float) -> None:
                 self.selected.append(dup.record)
         self.threshold = threshold
 
+    @property
+    def selected_with_duplicates(self) -> list[tuple[Record, list[tuple[Record, float]]]]:
+        """
+        For every kept record, return the duplicates that were removed along with their similarity scores.
+
+        :return: A list of tuples where each tuple contains a kept record
+                and a list of its duplicates with their similarity scores.
+        """
+
+        def _to_hashable(record: Record) -> Hashable:
```

Comment:
We already have a `to_hashable` (which doesn't work on strings though). So use that one instead of duplicating.

https://github.com/MinishLab/semhash/blob/1d8095f5c9da89c476dc8416532359c7b9a38d13/semhash/records.py#L8

Also, this code doesn't work if one of the values of your dict is not hashable, e.g.,:

```python
{"a": {1, 2, 3}}
```

is not hashable. So that requires a different solution.

Reply:
In the main code we circumvent this by only hashing based on specific columns.